File: 프로그래머스/1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.py
# 숫자 10 처리하려면 점수를 문자열로 이어붙어야함
def solution(dartResult):
    result = []
    score = ''
    for dart in dartResult:
        if dart.isdigit():
            score += dart
        elif dart == 'S':
            result.append(int(score))
            score = ''
        elif dart == 'D':
            result.append(int(score)**2)
            score = ''
        elif dart == 'T':
            result.append(int(score)**3)
            score = ''
        elif dart == '*':
            if len(result) <= 1:
                result[0] *= 2
            else:
                result[-2] *= 2
                result[-1] *= 2
        elif dart == '#':
            result[-1] *= -1
        
    return sum(result)

File: 프로그래머스/1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.js
function solution(dartResult) {
  let score = '';
  let result = [];
  [...dartResult].forEach(dart => {
    if(dart === 'S'){
      result.push(parseInt(score))
      score = ''
    } else if(dart === 'D'){
      result.push(parseInt(score) ** 2);
      score = ''
    } else if(dart === 'T'){
      result.push(parseInt(score) ** 3);
      score = ''
    } else if(dart === '*'){
      const lenResult = result.length
      if(lenResult <= 1) {
        result[0] *= 2;
      } else {
        result[lenResult-2] *= 2;
        result[lenResult-1] *= 2;
      }
    } else if(dart === '#'){
      result[result.length-1] *= -1;
    } else {
      score += dart;
    }
  })

  return result.reduce((acc, cur) => acc + cur, 0);
}